src/shaders/workingRef-vert.glsl

```diff
@@ -0,0 +1,238 @@
+// PRODUCES NOISE VALUE FOR POINT by interpolating surrounding lattice values (8 corner points)
+float interpNoiseByCorners(float x, float y, float z) {
+    float interval = 13.0;  
+
+    float xDist = x - mod(x, interval);
+    float yDist = y - mod(y, interval);
+    float zDist = z - mod(z, interval);
+
+    float xLow = x - xDist;
+    float yLow = y - yDist;
+    float zLow = z - zDist;
+
+    float xHigh = xLow + interval;
+    float yHigh = yLow + interval;
+    float zHigh = zLow + interval;
+
+    //upper cube check loc
+    float tl_up = currNoiseFunc(xLow,  yHigh, zHigh);
+    float tr_up = currNoiseFunc(xHigh, yHigh, zHigh);
+    float bl_up = currNoiseFunc(xLow,  yLow,  zHigh);
+    float br_up = currNoiseFunc(xHigh, yLow,  zHigh);
+    //lower cube check loc
+    float tl_bot = currNoiseFunc(xLow,  yHigh, zLow);
+    float tr_bot = currNoiseFunc(xHigh, yHigh, zLow);
+    float bl_bot = currNoiseFunc(xLow,  yLow,  zLow);
+    float br_bot = currNoiseFunc(xHigh, yLow,  zLow);
+    
+    // cant just divide all 
+    // interp1: need to interpolate between two of opp x with same y and same z loc (8 locations yield 4 locations)
+    // interp2: need to interpolate between two of x with opp y and same z loc (4 locations yield 2 locations)
+    // interp3: need to interpolate between two of x and y with opp z loc (2 locations yield 1 value)
+    
+    // interp value t set as .5 bc always directly halfway between the two locations
+    float t = 0.5;
+    
+    // interp1 //
+    float top_up = abs( cosInterpFunc(tl_up, tr_up, t) );
+    float back_up = abs( cosInterpFunc(bl_up, br_up, t) );
+    float top_bot = abs( cosInterpFunc(tl_bot, tr_bot, t) );
+    float back_bot = abs( cosInterpFunc(bl_bot, br_bot, t) );
+    
+    // interp2 //
+    float up = abs( cosInterpFunc(top_up, back_up, t) );
+    float bot = abs( cosInterpFunc(top_bot, back_bot, t) );
+    
+    // interp3 // TAKING ABS HERE NOW
+    float val = abs( cosInterpFunc(up, bot, t) );
+    
+    return val;
+}
+
+// AVERAGES NOISE RESULTS BASED ON SURROUNDINGS at an x,y,z location with neighboring values (x +/-1, y +/-1, z +/-1)
+// calculates overall noise at given location in func so don't have to input as param
+float smooth(float x, float y, float z) {
+    float ave = 0.0;
+    float count = 0.0;
+
+    float i = x - 1.0;
+    float j = y - 1.0;
+    float k = z - 1.0;
+    
+    for (int q = 0; q < 2; q++) {
+        for (int r = 0; r < 2; r++) {
+            for (int s = 0; s < 2; s++) {
+                ave += interpNoiseByCorners(i, j, k);
+                count += 1.0;
+
+                k++;
+            }
+            j++;
+        }
+        i++;
+    }
+    
+    return (ave / count);
+}
+
+// 3D MULTI-OCTAVE LATTICE-VALUE NOISE FUNCTION generates output in controlled range [0,1] or [-1,1]
+float overallNoiseFunc(float in1, float in2, float in3) {
+    in1 /= .002;
+    in2 /= .005;
+    in3 /= .007;
+
+    float total = 0.0;
+    float persistence = 1.2;
```

Comment:
Could be stylistic, but persistence greater than 1.0 will make your higher octaves become more and more powerful resulting in something probably a bit too noisy. Amplitude exponentially increases with persistence.